tests/test_commands.py

```diff
@@ -101,14 +101,27 @@ def run_bg(self, *command, script=None):
 
         yield proc
 
-        os.kill(proc.pid, signal.SIGINT)  # Ctrl+C
+        try:
+            # Shutdown the server via POST request
+            url = 'http://localhost:{}/__shutdown__/'.format(port)
+            print('Shutting down via', url)
+            requests.post(url)
+        except Exception as e:
+            print(e)
+            proc.kill()
+
```

Comment:
You could also:
* Before sending the request, check if the process isn't already dead.
* Put the cleanup in a `finally` block, with everything after getting `proc` up to the `yield` in the `try`